README.md

```diff
@@ -1,2 +1,246 @@
 # Bring
-Dependency Injection Framework
+
+---
+
+## What is Bring?
+
+---
+**Bring is a dependency injection framework**. It uses IoC (Inversion of Control) container, which is represented by
+[ApplicationContext](src/main/java/com/bobocode/hoverla/bring/context/ApplicationContext.java),
+where [beans](#bean) are managed. The framework can create beans for you and inject them as dependencies to other beans
+
+## Quick Start
+
+---
+To easily start an application with Bring, use the `BringApplication.loadContext("package.to.scan")` method
+
+```java
+import com.bobocode.hoverla.bring.BringApplication;
+import com.bobocode.hoverla.bring.context.ApplicationContext;
+
+public class DemoApp {
+    public static void main(String[] args) {
+        ApplicationContext = BringApplication.loadContext("package.to.scan"); // provide packages to be scanned for beans 
+    }
+}
+```
+
+If you need to customize loading of `ApplicationContext`, you can do it with `ApplicationContextBuilder`. Here's an
+example of its usage:
+
+```java
+import ch.qos.logback.classic.Level;
+import com.bobocode.hoverla.bring.BringApplication;
+import com.bobocode.hoverla.bring.context.ApplicationContext;
+
+public class DemoApp {
+    public static void main(String[] args) {
+        ApplicationContext = BringApplication.getContextBuilder()
+                .logLevel(Level.DEBUG)                 // provide logging level
+                .packagesToScan("package.to.scan")     // provide packages to be scanned for beans
+                .build();
+    }
+}
+```
+
+### Bean
+
+---
+**Bean** is a class that is managed by IoC container. To define a bean you have multiple options:
+
+- [Annotation Configuration](#annotation-configuration)
+- [Java Configuration](#java-configuration)
+
+### Annotation configuration
+
+---
+To define a bean with **annotation configuration** just annotate your desired bean class with
+[@Bean](src/main/java/com/bobocode/hoverla/bring/annotation/Bean.java) annotation. Look at the next example:
+
+```java
+import com.bobocode.hoverla.bring.annotation.Bean;
+
+@Bean
+public class MySimpleBean {
+
+}
+```
+
+By default, **Bring will use a class name as a bean name**. You can customize it by providing a `name` property for
+the [@Bean](src/main/java/com/bobocode/hoverla/bring/annotation/Bean.java) annotation
+
+```java
+import com.bobocode.hoverla.bring.annotation.Bean;
+
+@Bean("customBean")
+public class MyCustomBean() {
+
+}
+```
+
+If your class has dependencies that should be injected, you may use one of these types of injection:
+
+- [Constructor Injection](#constructor-injection)
+- [Field Injection](#field-injection)
+
+A bean class has the following constraints:
+
+- only one constructor with fields should be
```

Comment:
It's outdated info, if class has several constructors then one of them must be marked with Inject. 
If there's only one constructor, it's not required to be marked

Reply:
fixed